src/bids/layout/tests/conftest.py

```diff
@@ -101,3 +102,10 @@ def layout_synthetic(tests_dir, request, db_dir):
 def layout_synthetic_nodb(tests_dir, request, db_dir):
     path = tests_dir / 'data' / 'synthetic'
     return BIDSLayout(path, derivatives=True)
+
+
+@pytest.fixture
+def temporary_dataset(tmp_path, tests_dir):
+    path = tests_dir / 'data' / 'ds005'
+    shutil.copytree(path, tmp_path / 'ds005')
+    return tmp_path / 'ds005'
```

Comment:
```suggestion
    yield tmp_path / 'ds005'
    shutil.rmtree(tmp_path / 'ds005')
```

Reply:
Is this a concern? `rmtree` is already run by the `tmp_path` fixture.

Reply:
Ah, TIL!

Reply:
https://github.com/pytest-dev/pytest/blob/150cac857d88c6a733d0f0f7723bf0223f4b3e22/src/_pytest/tmpdir.py#L254-L277